src/main/java/core/basesyntax/FileReaderImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> input(String path) {
+        try{
+            Files.readAllLines(Paths.get(path));
+
+            return Files.readAllLines(Paths.get(path));
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file: " + path + e);
```

Catching IOException and then concatenating the exception object to the message loses the original exception as the cause and makes debugging harder. This violates the checklist item: "Do not ignore exceptions in try-catch blocks; do not leave `e.printStackTrace()` or `System.out.println(e.getMessage())` in catch blocks. Rethrow a `RuntimeException` with an informative message and the exception object." Use `throw new RuntimeException("Can't read data from file: " + path, e);` so the original exception is preserved as the cause.